src/main/java/com/manardenza/service/ReservationService.java

```diff
@@ -0,0 +1,54 @@
+package com.manardenza.service;
+
+import com.manardenza.dao.ReservationDaoImpl;
+import com.manardenza.entity.Reservation;
+import com.manardenza.entity.Room;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public final class ReservationService {
+
+    private static ReservationService instance;
+    private ReservationDaoImpl reservationDao;
+
+    private ReservationService() {
+        this.reservationDao = ReservationDaoImpl.getInstance();
+    }
+
+    public static ReservationService getInstance() {
+        if (instance == null) {
+            instance = new ReservationService();
+        }
+        return instance;
+    }
+
+    public Long bookRoom(Reservation reservation) {
+        return reservationDao.save(reservation).getId();
+    }
+
+    public boolean cancelReservation(long idReservation) {
+        return reservationDao.getAll().removeIf(p -> p.getId() == idReservation);
+    }
+
+    public List<Room> checkRoomReservation(Date reservedFrom, Date reservedTo, List<Room> findRooms) {
+        List<Room> toDel = new ArrayList<>();
+        for (Room findRoom : findRooms) {
+            if (reservationDao.getAll().stream().filter(p -> p.getReservedRoom().equals(findRoom)
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (10) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1067)